### PR TITLE
feat: add networks and remove self link

### PR DESCRIPTION
**Pull request by MiguelAHM**

### CODE REVIEW

**Comment by MiguelAHM** [comment]
https://gitlab.ripe.net/swe-database-team/whois/-/pipelines/282630

Test ok

File: whois-api/src/main/java/net/ripe/db/whois/api/rdap/WhoisRdapService.java [hunk 1]
```diff
@@ -290,6 +301,28 @@ private Response getOrganisationResponse(final HttpServletRequest request,
                 .header(CONTENT_TYPE, CONTENT_TYPE_RDAP_JSON)
                 .build();
     }
+
+    private Response getDomainResponse(HttpServletRequest request, Iterable<RpslObject> domainResult,
+                                       Iterable<RpslObject> inetnumResult) {
+        final Iterator<RpslObject> domainIterator = domainResult.iterator();
```
Comment (eshryane) [inline whois-api/src/main/java/net/ripe/db/whois/api/rdap/WhoisRdapService.java:307]: Aha! Arguments should be Iterator

File: whois-api/src/test/java/net/ripe/db/whois/api/elasticsearch/WhoisRdapElasticServiceTestIntegration.java [hunk 2]
```diff
@@ -695,19 +699,6 @@ private void assertCommon(RdapObject object) {
...
 
-    private void assertCopyrightLink(final List<Link> links, final String value) {
-        assertThat(links, hasSize(2));
-        Collections.sort(links);
-
-        assertThat(links.get(0).getRel(), is("copyright"));
-        assertThat(links.get(0).getHref(), is("http://www.ripe.net/data-tools/support/documentation/terms"));
-        assertThat(links.get(0).getHref(), is("http://www.ripe.net/data-tools/support/documentation/terms"));
-
-        assertThat(links.get(1).getRel(), is("self"));
```
Comment (eshryane) [inline whois-api/src/test/java/net/ripe/db/whois/api/elasticsearch/WhoisRdapElasticServiceTestIntegration.java:706]: Why were there two elements in `links` previously and only one now?

File: whois-api/src/test/java/net/ripe/db/whois/api/rdap/WhoisRdapServiceTestIntegration.java [hunk 3]
```diff
@@ -910,9 +912,163 @@ public void lookup_domain_object() {
         assertThat(notices.get(1).getTitle(), is("Source"));
         assertTnCNotice(notices.get(2), "https://rdap.db.ripe.net/domain/31.12.202.in-addr.arpa");
 
-        assertCopyrightLink(domain.getLinks(), "https://rdap.db.ripe.net/domain/31.12.202.in-addr.arpa");
+        final List<Link> links = domain.getLinks();
+        assertThat(links.size(), is(1));
```
Comment (eshryane) [inline whois-api/src/test/java/net/ripe/db/whois/api/rdap/WhoisRdapServiceTestIntegration.java:916]: use `hasSize(1)` for consistency

File: whois-api/src/test/java/net/ripe/db/whois/api/rdap/WhoisRdapServiceTestIntegration.java [hunk 4]
```diff
@@ -910,9 +912,163 @@ public void lookup_domain_object() {
...
+                .get(Domain.class);
+
+        assertCommon(domain);
+        assertThat(domain.getHandle(), equalTo("52.179.80.in-addr.arpa"));
+        assertThat(domain.getLdhName(), equalTo("52.179.80.in-addr.arpa."));
+        assertThat(domain.getObjectClassName(), is("domain"));
+        assertThat(domain.getNetwork().getHandle(), is("80.179.52.0 - 80.179.55.255"));
+
+        final List<Link> links = domain.getLinks();
+        assertThat(links.size(), is(1));
```
Comment (eshryane) [inline whois-api/src/test/java/net/ripe/db/whois/api/rdap/WhoisRdapServiceTestIntegration.java:967]: Use `hasSize(1)`

File: whois-api/src/test/java/net/ripe/db/whois/api/rdap/WhoisRdapServiceTestIntegration.java [hunk 5]
```diff
@@ -910,9 +912,163 @@ public void lookup_domain_object() {
...
+                .get(Domain.class);
+
+        assertCommon(domain);
+        assertThat(domain.getHandle(), equalTo("e.c.c.2.0.0.a.2.ip6.arpa"));
+        assertThat(domain.getLdhName(), equalTo("e.c.c.2.0.0.a.2.ip6.arpa."));
+        assertThat(domain.getObjectClassName(), is("domain"));
+        assertThat(domain.getNetwork().getHandle(), is("2a00:2cce::/32"));
+
+        final List<Link> links = domain.getLinks();
+        assertThat(links.size(), is(1));
```
Comment (eshryane) [inline whois-api/src/test/java/net/ripe/db/whois/api/rdap/WhoisRdapServiceTestIntegration.java:1018]: Use `hasSize(1)`

File: whois-api/src/test/java/net/ripe/db/whois/api/rdap/WhoisRdapServiceTestIntegration.java [hunk 6]
```diff
@@ -910,9 +912,163 @@ public void lookup_domain_object() {
...
+                .get(Domain.class);
+
+        assertCommon(domain);
+        assertThat(domain.getHandle(), equalTo("52.179.80.in-addr.arpa"));
+        assertThat(domain.getLdhName(), equalTo("52.179.80.in-addr.arpa."));
+        assertThat(domain.getObjectClassName(), is("domain"));
+        assertNull(domain.getNetwork());
+
+        final List<Link> links = domain.getLinks();
+        assertThat(links.size(), is(1));
```
Comment (eshryane) [inline whois-api/src/test/java/net/ripe/db/whois/api/rdap/WhoisRdapServiceTestIntegration.java:1068]: use `hasSize(1)`

File: whois-api/src/test/java/net/ripe/db/whois/api/rdap/WhoisRdapServiceTestIntegration.java [hunk 7]
```diff
@@ -923,6 +1079,12 @@ public void lookup_domain_object_is_case_insensitive() {
         assertThat(domain.getHandle(), equalTo("31.12.202.in-addr.arpa"));
         assertThat(domain.getLdhName(), equalTo("31.12.202.in-addr.arpa."));
         assertThat(domain.getObjectClassName(), is("domain"));
+        assertThat(domain.getNetwork().getHandle(), is("0.0.0.0 - 255.255.255.255"));
+
+        final List<Link> links = domain.getLinks();
+        assertThat(links.size(), is(1));
```
Comment (eshryane) [inline whois-api/src/test/java/net/ripe/db/whois/api/rdap/WhoisRdapServiceTestIntegration.java:1085]: Use `hasSize(1)`

File: whois-api/src/test/java/net/ripe/db/whois/api/rdap/WhoisRdapServiceTestIntegration.java [hunk 8]
```diff
@@ -910,9 +912,163 @@ public void lookup_domain_object() {
...
+
+        final Domain domain = createResource("domain/52.179.80.in-addr.arpa")
+                .request(MediaType.APPLICATION_JSON_TYPE)
+                .get(Domain.class);
+
+        assertCommon(domain);
+        assertThat(domain.getHandle(), equalTo("52.179.80.in-addr.arpa"));
+        assertThat(domain.getLdhName(), equalTo("52.179.80.in-addr.arpa."));
+        assertThat(domain.getObjectClassName(), is("domain"));
+        assertNull(domain.getNetwork());
```
Comment (eshryane) [inline whois-api/src/test/java/net/ripe/db/whois/api/rdap/WhoisRdapServiceTestIntegration.java:1065]: Use `assertThat(domain.getNetwork(), is(nullValue());` for consistency

File: whois-api/src/test/java/net/ripe/db/whois/api/elasticsearch/WhoisRdapElasticServiceTestIntegration.java (hunk 2 above)
Comment (MiguelAHM) [reply whois-api/src/test/java/net/ripe/db/whois/api/elasticsearch/WhoisRdapElasticServiceTestIntegration.java:706]: we are removing self, it is specified in jira ticket

File: whois-api/src/main/java/net/ripe/db/whois/api/rdap/WhoisRdapService.java [hunk 9]
```diff
@@ -219,6 +220,16 @@ private Boolean isRedirect(ObjectType objectType, final String key) {
         return !delegatedStatsService.isMaintainedInRirSpace(source.getName(), objectType, CIString.ciString(key));
     }
 
+    protected Response lookupForDomain(final HttpServletRequest request, final String key) {
+        final Domain domain = Domain.parse(key);
+        final Iterable<RpslObject> domainResult =
+                rdapQueryHandler.handleQueryStream(getQueryObject(ImmutableSet.of(ObjectType.DOMAIN),
+                        key), request).collect(Collectors.toList());
+        final Iterable<RpslObject> inetnumResult =
```
[the remark this answers is not in the document]
Comment (MiguelAHM) [reply whois-api/src/main/java/net/ripe/db/whois/api/rdap/WhoisRdapService.java:228]: As commented I will be working directly with the Stream

**Review by eshryane — APPROVED** [review]

**Pull request merged**